libs/providers/launchdarkly-client/src/lib/launchdarkly-client-provider.ts

```diff
@@ -29,78 +32,114 @@ function wrongTypeResult<T>(value: T): ResolutionDetails<T> {
   };
 }
 
+/*
+ * Provider Status
+ * defined here https://github.com/open-feature/js-sdk/blob/d26c4eeedde9cc321c69375594f1ccbfaff41a28/packages/shared/src/types.ts#L265
+ * ideally will be exported by '@openfeature/web-sdk'
+ * */
+enum ProviderStatus {
+  NOT_READY = 'NOT_READY',
+  READY = 'READY',
+  ERROR = 'ERROR',
+}
+
 export class LaunchDarklyClientProvider implements Provider {
   readonly metadata: ProviderMetadata = {
-    name: 'launchdarkly-client-provider',
+    name: 'launchdarkly-client',
   };
 
+  private readonly ldOptions: LDOptions | undefined;
   private readonly logger: Logger;
+  private _client?: LDClient;
+
+  public events = new OpenFeatureEventEmitter();
 
-  constructor(private client: LDClient, options: LaunchDarklyProviderOptions = {}) {
-    if (options.logger) {
-      this.logger = options.logger;
+  /*
+   * implement status field/accessor
+   * https://openfeature.dev/specification/sections/providers#requirement-242
+   * */
+  public status: ProviderStatus = ProviderStatus.NOT_READY;
+
+  constructor(private readonly envKey: string, { providerLogger, ...ldOptions }: LaunchDarklyProviderOptions) {
+    this.ldOptions = ldOptions;
+    if (providerLogger) {
+      this.logger = providerLogger;
     } else {
       this.logger = basicLogger({ level: 'info' });
     }
   }
 
-  async initialize(context: EvaluationContext): Promise<void> {
-    const newContext = this.translateContext(context);
-    const oldContext = this.client.getContext();
-    const ignoreAnonymous = newContext.anonymous && oldContext.anonymous && !newContext.key
-    if(!ignoreAnonymous && !isEqual(newContext, oldContext)) {
-       await this.client.identify(newContext);
+  private get client(): LDClient {
+    if (!this._client) {
+      throw new GeneralError('Provider is not initialized');
+    }
+    return this._client;
+  }
+
+  async initialize(): Promise<void> {
+    /* for simplicity always initialize with anonymous by default*/
+
+    this._client = initialize(this.envKey, { anonymous: true }, this.ldOptions);
+
+    /*
+     * set event listeners to propagate ld events to open feature provider
+     * */
+    this.setListeners();
```

Comment:
Perhaps this should be optional, based on a configuration?

I'm not sure if all users would make use of this or not, or if there's performance/traffic costs associated with setting up such listeners in LD.